helmclient.go

```diff
@@ -574,85 +597,107 @@ func (c *Client) ListReleaseContents(ctx context.Context) ([]*ReleaseContent, er
 
 		contents = append(contents, content)
 	}
-
+	t.ObserveDuration()
 	return contents, nil
 }
 
 // LoadChart loads a Helm Chart and returns relevant parts of its structure.
-func (c *Client) LoadChart(ctx context.Context, chartPath string) (Chart, error) {
+func (c *Client) LoadChart(ctx context.Context, chartPath string) (chart Chart, err error) {
+	eventName := "load_chart"
+	defer errorReporting(eventName, err)
+	t := prometheus.NewTimer(controllerHistogram.WithLabelValues(eventName))
+
 	helmChart, err := chartutil.Load(chartPath)
 	if err != nil {
 		return Chart{}, microerror.Mask(err)
 	}
 
-	chart, err := newChart(helmChart)
+	chart, err = newChart(helmChart)
 	if err != nil {
 		return Chart{}, microerror.Mask(err)
 	}
-
+	t.ObserveDuration()
 	return chart, nil
 }
 
 // PingTiller proxies the underlying Helm client PingTiller method.
-func (c *Client) PingTiller(ctx context.Context) error {
-	t, err := c.newTunnel()
+func (c *Client) PingTiller(ctx context.Context) (err error) {
+	eventName := "ping_tiller"
+	defer errorReporting(eventName, err)
+	t := prometheus.NewTimer(controllerHistogram.WithLabelValues(eventName))
+
+	tunnel, err := c.newTunnel()
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	defer c.closeTunnel(ctx, t)
+	defer c.closeTunnel(ctx, tunnel)
 
-	err = c.newHelmClientFromTunnel(t).PingTiller()
+	err = c.newHelmClientFromTunnel(tunnel).PingTiller()
 	if err != nil {
 		return microerror.Mask(err)
 	}
-
+	t.ObserveDuration()
 	return nil
 }
 
 // RunReleaseTest runs the tests for a Helm Release. The releaseName is the
 // name of the Helm Release that is set when the Helm Chart is installed. This
 // is the same action as running the helm test command.
-func (c *Client) RunReleaseTest(ctx context.Context, releaseName string, options ...helmclient.ReleaseTestOption) error {
+func (c *Client) RunReleaseTest(ctx context.Context, releaseName string, options ...helmclient.ReleaseTestOption) (err error) {
+	eventName := "run_release_test"
+	defer errorReporting(eventName, err)
+	t := prometheus.NewTimer(controllerHistogram.WithLabelValues(eventName))
+
 	c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("running tests for release %#q", releaseName))
 
-	t, err := c.newTunnel()
+	tunnel, err := c.newTunnel()
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	defer c.closeTunnel(ctx, t)
+	defer c.closeTunnel(ctx, tunnel)
 
-	resChan, errChan := c.newHelmClientFromTunnel(t).RunReleaseTest(releaseName, helmclient.ReleaseTestTimeout(int64(runReleaseTestTimout)))
+	resChan, errChan := c.newHelmClientFromTunnel(tunnel).RunReleaseTest(releaseName, helmclient.ReleaseTestTimeout(int64(runReleaseTestTimout)))
 	if IsReleaseNotFound(err) {
 		return backoff.Permanent(microerror.Mask(err))
 	} else if err != nil {
 		return microerror.Mask(err)
 	}
 
+W:
 	for {
 		select {
 		case err := <-errChan:
 			if err != nil {
-				return microerror.Mask(err)
+				err = microerror.Mask(err)
+				break W
```

Comment:
Why is this `break W`?

Reply:
Otherwise, it would only break inner `select` statement.